db/migrate/20210428122404_add_company_to_product.rb

```diff
@@ -0,0 +1,9 @@
+class AddCompanyToProduct < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :products, :company, foreign_key: true
+    Product.all.each do |product|
+      product.update_attribute :company_id, product.customer.company_id
+    end
```

Comment:
Se fosse muitos dados aqui seria legal fazer in batches pra não sobrecarregar a memória, também seria legal usar uma transaction pra falhar a migration se a persistência não acontecer:

```suggestion
    Product.transaction do
      Product.find_each do |product|
        product.update_attribute :company_id, product.customer.company_id
      end
    end
```